lib/improver/spotdata/extract_data.py

```diff
@@ -157,52 +159,111 @@ def process(self, cube, sites, neighbours, ancillary_data, additional_data,
+                             'coord_type': AuxCoord},
+                'longitude': {'units': 'degrees', 'data_type': float,
+                              'coord_type': AuxCoord},
+                'altitude': {'units': 'm', 'data_type': float,
+                             'coord_type': AuxCoord,
+                             'custom_function': np.nan_to_num},
+                'wmo_site': {'data_type': int, 'coord_type': AuxCoord},
+                'utc_offset': {'units': 'hours', 'data_type': float,
+                               'coord_type': AuxCoord}
+                }
 
-        utc_offsets : list
-            A list of UTC off sets in hours ordered to match the sites
-            OrderedDict.
+    @staticmethod
+    def make_stat_coordinate_first(cube):
+        """
+        Reorder cube dimension coordinates to ensure the statistical
+        coordinate is first.
 
-        wmo_sites : list
-            A list of flags indicating whether the site is associated with a
-            wmo observing site (1) or is not associated (0).
+        Args:
+        -----
+        cube : iris.cube.Cube
+        The cube to be reordered.
 
         Returns:
         --------
-        Creates iris.DimCoord and iris.AuxCoord objects from the provided data
-        for use in constructing new cubes.
+        cube : iris.cube.Cube
+        Cube with the statistical coordinate moved to be first.
+
+        Raises:
+        -------
+        Warning if more than one statistical dimension is found. Then promotes
+        the first found to become the leading dimension.
 
         """
-        indices = DimCoord(np.arange(len(latitudes)), long_name='index',
-                           units='1')
-        latitude = AuxCoord(latitudes, standard_name='latitude',
-                            units='degrees')
-        longitude = AuxCoord(longitudes, standard_name='longitude',
-                             units='degrees')
-        altitude = AuxCoord(altitudes, standard_name='altitude',
-                            units='m')
-        utc_offset = AuxCoord(utc_offsets, long_name='utc_offset',
-                              units='hours')
-        wmo_site = AuxCoord(wmo_sites, long_name='wmo_site', units='1')
-
-        return indices, latitude, longitude, altitude, utc_offset, wmo_site
+        stat_coordinates = ['realization', 'percentile_over']
+        cube_dimension_order = {
+            coord.name(): cube.coord_dims(coord.name())[0]
+            for coord in cube.dim_coords}
+
+        stat_coord = []
+        for crd in stat_coordinates:
+            stat_coord += [coord for coord in cube_dimension_order.keys()
+                           if crd in coord]
+
+        if len(stat_coord) > 1:
+            stat_coord = stat_coord[0]
+            msg = ('More than one statistical coordinate found. Promoting the '
+                   'first found, {}, to the leading dimension.'.format(
+                    stat_coord))
+            warnings.warn(msg)
+
```

Comment:
I think that it is possible to have multiple statistical coordinates, for example, `realization` and `percentiles_over_neighbourhood`, as the two statistical coordinates are not interdependent. I don't think it would be possible to have `realization` and `percentiles_over_realization` on the same cube though.